tests/Action/SetObjectFieldValueActionTest.php

```diff
@@ -344,14 +367,130 @@ public function testSetObjectFieldEditableMultipleValue(): void
-        $fieldDescription->getType()->willReturn('boolean');
-        $fieldDescription->getTemplate()->willReturn('field_template');
+        $fieldDescription->getType()->willReturn(null);
+        $fieldDescription->getTemplate()->willReturn(false);
         $fieldDescription->getValue(Argument::cetera())->willReturn(['some value']);
+        $fieldDescription->getOption('data_transformer')->willReturn(null);
+
+        $this->validator->validate($object)->willReturn(new ConstraintViolationList([]));
+
+        $response = ($this->action)($request);
+
+        $this->assertSame([1, 2], $object->status);
+        $this->assertSame(Response::HTTP_OK, $response->getStatusCode());
+    }
+
+    public function testSetObjectFieldTransformed(): void
+    {
+        $object = new Foo();
+        $request = new Request([
+            'code' => 'sonata.post.admin',
+            'objectId' => 42,
+            'field' => 'enabled',
+            'value' => 'yes',
+            'context' => 'list',
+        ], [], [], [], [], ['REQUEST_METHOD' => Request::METHOD_POST, 'HTTP_X_REQUESTED_WITH' => 'XMLHttpRequest']);
+
+        $data_transformer = new CallbackTransformer(static function ($value) {
+            return (string) (int) $value;
+        }, static function ($value) {
+            return filter_var($value, FILTER_VALIDATE_BOOLEAN);
+        });
+
+        $fieldDescription = $this->prophesize(FieldDescriptionInterface::class);
+        $pool = $this->prophesize(Pool::class);
+        $translator = $this->prophesize(TranslatorInterface::class);
+        $propertyAccessor = new PropertyAccessor();
+        $templateRegistry = $this->prophesize(TemplateRegistryInterface::class);
+        $container = $this->prophesize(ContainerInterface::class);
+
+        $this->admin->getObject(42)->willReturn($object);
+        $this->admin->getCode()->willReturn('sonata.post.admin');
+        $this->admin->hasAccess('edit', $object)->willReturn(true);
+        $this->admin->getListFieldDescription('enabled')->willReturn($fieldDescription->reveal());
+        $this->admin->update($object)->shouldBeCalled();
+        // NEXT_MAJOR: Remove this line
+        $this->admin->getTemplate('base_list_field')->willReturn('admin_template');
+        $templateRegistry->getTemplate('base_list_field')->willReturn('admin_template');
+        $container->get('sonata.post.admin.template_registry')->willReturn($templateRegistry->reveal());
+        $this->pool->getPropertyAccessor()->willReturn($propertyAccessor);
+        $this->twig->addExtension(new SonataAdminExtension(
+            $pool->reveal(),
+            null,
+            $translator->reveal(),
+            $container->reveal()
+        ));
+        $fieldDescription->getOption('editable')->willReturn(true);
+        $fieldDescription->getAdmin()->willReturn($this->admin->reveal());
+        $fieldDescription->getType()->willReturn(null);
+        $fieldDescription->getTemplate()->willReturn(false);
```

Comment:
Could we return `null`?

Reply:
Using `null` as the return value for the `getTemplate()` method does not break the tests.